doc/classes/Decal.xml

```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="Decal" inherits="VisualInstance3D" version="4.0">
+	<brief_description>
+		Node that projects a texture onto a [MeshInstance3D].
+	</brief_description>
+	<description>
+		[Decal]s are used to project a texture onto a [Mesh] in the scene. Use Decals to add detail to a scene without affecting the underlying [Mesh]. They are often used to add weathering to building, add dirt or mud to the ground, or add variety to props.
+		They are made of an [AABB] and a group of [Texture2D]s specifying [Color], normal, ORM (ambient occlusion, roughness, metallic), and emission. Decals are projected within their [AABB] so altering the orientation of the Decal affects the direction in which they are projected. By default, Decals are projected down (i.e. from positive Y to negative Y).
+		The [Texture2D]s associated with the Decal are automatically stored in a texture atlas which is used for drawing the decals so all decals can be drawn at once. Godot uses clustered decals, meaning they are stored in cluster data and drawn when the mesh is drawn, they are not drawn as a postprocessing effect after.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="get_texture" qualifiers="const">
+			<return type="Texture2D">
+			</return>
+			<argument index="0" name="type" type="int" enum="Decal.DecalTexture">
+			</argument>
+			<description>
+				Returns the [Texture2D] associated with the specified [enum DecalTexture]. This is a convenience method, in most cases you should access the texture directly. 
+				For example, instead of [code]albedo_tex = $Decal.get_texture(Decal.TEXTURE_ALBEDO)[/code], use [code]albedo_tex = $Decal.texture_albedo[/code].
+				One case where this is better than accessing the texture directly is when you want to copy one Decal's textures to another. For example:
+				[codeblock]
+				for i in range(Decal.TEXTURE_MAX):
```

Comment:
Is this `range()` necessary?

Reply:
I guess not! I didn't realize you could iterate over an enum without it.